tenant/http_server_user.go

```diff
@@ -495,3 +446,87 @@ func decodePatchUserRequest(ctx context.Context, r *http.Request) (*patchUserReq
 		UserID: i,
 	}, nil
 }
+
+// MeHandler represents an HTTP API handler for /me routes.
+type MeHandler struct {
+	chi.Router
+	api         *kithttp.API
+	log         *zap.Logger
+	userSvc     influxdb.UserService
+	passwordSvc influxdb.PasswordsService
+}
+
+func (h *MeHandler) Prefix() string {
+	return prefixMe
+}
+
+func NewHTTPMeHandler(log *zap.Logger, userService influxdb.UserService, passwordService influxdb.PasswordsService) *MeHandler {
+	svr := &MeHandler{
+		api:         kithttp.NewAPI(kithttp.WithLog(log)),
+		log:         log,
+		userSvc:     userService,
+		passwordSvc: passwordService,
+	}
+
+	r := chi.NewRouter()
+	r.Use(
+		middleware.Recoverer,
+		middleware.RequestID,
+		middleware.RealIP,
+	)
+
+	// RESTy routes for "articles" resource
+	r.Route("/", func(r chi.Router) {
+		r.Get("/", svr.handleMe)
+		r.Put("/password", svr.handlePutMePassword)
+	})
+
+	svr.Router = r
+	return svr
+}
+
+func (h *MeHandler) getUserID(ctx context.Context) (*platform.ID, error) {
+	a, err := icontext.GetAuthorizer(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	id := a.GetUserID()
+	return &id, nil
+}
+
+func (h *MeHandler) handleMe(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	userID, err := h.getUserID(ctx)
+	if err != nil {
+		h.api.Err(w, r, err)
+	}
+	user, err := h.userSvc.FindUserByID(ctx, *userID)
+	if err != nil {
+		h.api.Err(w, r, err)
+	}
+	h.api.Respond(w, r, http.StatusOK, newUserResponse(user))
```

Comment:
On error, does this respond with the err, then the StatusOk?  Should there be a return after the `h.Api.Err`  calls?

Reply:
You are correct, it is missing the return. Must have missed that when I was splicing code around.

Reply:
There are quite a few of those in the code, not just in this method.